src/main.rs

```diff
@@ -265,6 +265,29 @@ fn main() -> Result<()> {
                     Ok(None) => Ok(()),
                     Err(e) => Err(e),
                 }
+            } else if let Some(execute) = &binary_args.execute {
```

Comment:
I don't think you need to repeat the whole branch code. Instead, in the `else` branch, you could just set the prerun argument to `Some(code)` or `None` based on the `-e` flag.